src/IssueViz.Security/HotspotsList/TableDataSource/HotspotTableEntryFactory.cs

```diff
@@ -0,0 +1,61 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.ComponentModel.Composition;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.TableManager;
+using SonarLint.VisualStudio.IssueVisualization.Models;
+using SonarLint.VisualStudio.IssueVisualization.Security.Models;
+
+namespace SonarLint.VisualStudio.IssueVisualization.Security.HotspotsList.TableDataSource
+{
+    internal interface IHotspotTableEntryFactory
+    {
+        ITableEntry Create(IAnalysisIssueVisualization issueVisualization);
+    }
+
+    [Export(typeof(IHotspotTableEntryFactory))]
+    internal class HotspotTableEntryFactory : IHotspotTableEntryFactory
+    {
+        private readonly IServiceProvider serviceProvider;
+
+        [ImportingConstructor]
+        public HotspotTableEntryFactory([Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider)
```

Comment:
Created for a better isolation between HotspotsTableDataSource and HotspotsTableEntry. This way HotspotsTableDataSource doesn't need to know that HotspotsTableEntry even exists and is not affected by its dependency changes (e.g. the new IVsUIShell2)